.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,29 @@
+name: pre-commit checks
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```

Comment:
Thanks for this @aniketpati1121!
Maybe it would be easier to just add this job to the `test-python` script, similar to Trainer: https://github.com/kubeflow/trainer/blob/master/.github/workflows/test-python.yaml#L7-L12